contracts/ProtocolParameters.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+/**
+ * @title ProtocolParameters
+ * @notice This contract defines the protocol fees and address in which that fees will be recieved.
+ */
+contract ProtocolParameters {
+    address public protocolFeeRecipient;
+    uint256 public protocolFeeRate;
+
+    modifier onlyProtocolFeeRecipient() {
+        require(msg.sender == protocolFeeRecipient, "Caller is not the protocol fee recipient");
+        _;
+    }
+
+    modifier lessThan5Percent(uint256 feeRate){
+        require(feeRate <= 500, "Fee rate must be between 0 and 500 (in basis points)");
+        _;
+    }
+
+    modifier nonZeroAddress(address addr) {
+        require(addr != address(0), "Address cannot be zero");
+        _;
+    }
+
+    constructor(address _protocolFeeRecipient, uint256 _protocolFeeRate){
+        require(_protocolFeeRecipient != address(0), "Invalid fee recipient address");
+        require(_protocolFeeRate <= 500, "Fee rate must be between 0 and 500 (in basis points)"); //0 to 5% , rate = 0.0001 * _protocolFeeRate;
+        protocolFeeRecipient = _protocolFeeRecipient;
+        protocolFeeRate = _protocolFeeRate;
+    }
+
+    function updateFeeRate(uint256 newFeeRate) external onlyProtocolFeeRecipient lessThan5Percent(newFeeRate) {
+        protocolFeeRate = newFeeRate;
+    }
+
+    function updateFeeRecipient(address newFeeRecipient) external onlyProtocolFeeRecipient nonZeroAddress(newFeeRecipient) {
```

Comment:
I have realized that, although right now the fee recipient will be an EOA (Externally Owned Account), in the future it could be a smart contract. In this later case, it would be difficult for the protocol fee recipient to call this function. Perhaps we should have another address, let's call it "manager", who would be different from the fee recipient, would always be an EOA and would have the authority to call the update functions.